dev-infrastructure/templates/aks-development.bicep

```diff
@@ -217,36 +244,46 @@ resource aksCluster 'Microsoft.ContainerService/managedClusters@2024-01-01' = {
     }
   }
   properties: {
+    aadProfile: {
+      managed: true
+      enableAzureRBAC: true
+    }
     nodeResourceGroup: '${resourceGroup().name}-aks1'
     apiServerAccessProfile: {
       enablePrivateCluster: enablePrivateCluster
     }
     addonProfiles: {
       azureKeyvaultSecretsProvider: {
         enabled: true
+        config: {
+          enableSecretRotation: 'true'
+          rotationPollInterval: '24h'
+          syncSecret: 'true'
+        }
       }
     }
     kubernetesVersion: kubernetesVersion
     enableRBAC: true
     dnsPrefix: dnsPrefix
     agentPoolProfiles: [
       {
-        name: 'systempool'
+        name: 'system'
         osType: 'Linux'
+        osSKU: 'AzureLinux'
         mode: 'System'
         orchestratorVersion: kubernetesVersion
-        enableAutoScaling: nodePoolEnableAutoScaling
+        enableAutoScaling: true
         enableEncryptionAtHost: true
         enableFIPS: true
         osDiskType: 'Ephemeral'
         osDiskSizeGB: osDiskSizeGB
-        count: agentCount
+        count: agentMinCount
         minCount: agentMinCount
         maxCount: agentMaxCount
         vmSize: agentVMSize
         vnetSubnetID: aksNodeSubnet.id
         podSubnetID: aksPodSubnet.id
-        maxPods: maxPods
+        maxPods: 100
```

Comment:
Make sense to keep max pods to default `100` for dev env, but recommended to set 250 for production along with dedicated infra agentpool. 

Otherwise LGTM.